vault/resource_pki_secret_backend_role_test.go

```diff
@@ -7,11 +7,98 @@ import (
+		resource.TestCheckResourceAttr(resourceName, "key_type", "rsa"),
+		resource.TestCheckResourceAttr(resourceName, "key_bits", "2048"),
+		resource.TestCheckResourceAttr(resourceName, "email_protection_flag", "false"),
+		resource.TestCheckResourceAttr(resourceName, "email_protection_flag", "false"),
+		resource.TestCheckResourceAttr(resourceName, "key_usage.#", "3"),
+		resource.TestCheckResourceAttr(resourceName, "key_usage.0", "DigitalSignature"),
+		resource.TestCheckResourceAttr(resourceName, "key_usage.1", "KeyAgreement"),
+		resource.TestCheckResourceAttr(resourceName, "key_usage.2", "KeyEncipherment"),
+		resource.TestCheckResourceAttr(resourceName, "ext_key_usage.#", "0"),
+		resource.TestCheckResourceAttr(resourceName, "use_csr_common_name", "true"),
+		resource.TestCheckResourceAttr(resourceName, "use_csr_sans", "true"),
+		resource.TestCheckResourceAttr(resourceName, "ou.0", "test"),
+		resource.TestCheckResourceAttr(resourceName, "organization.0", "test"),
+		resource.TestCheckResourceAttr(resourceName, "country.0", "test"),
+		resource.TestCheckResourceAttr(resourceName, "locality.0", "test"),
+		resource.TestCheckResourceAttr(resourceName, "province.0", "test"),
+		resource.TestCheckResourceAttr(resourceName, "street_address.0", "123 test"),
+		resource.TestCheckResourceAttr(resourceName, "postal_code.0", "12345"),
+		resource.TestCheckResourceAttr(resourceName, "generate_lease", "false"),
+		resource.TestCheckResourceAttr(resourceName, "no_store", "false"),
+		resource.TestCheckResourceAttr(resourceName, "require_cn", "true"),
+		resource.TestCheckResourceAttr(resourceName, "basic_constraints_valid_for_non_ca", "false"),
+		resource.TestCheckResourceAttr(resourceName, "not_before_duration", "45m"),
+	}
+	resource.Test(t, resource.TestCase{
+		Providers:    testProviders,
+		PreCheck:     func() { testutil.TestAccPreCheck(t) },
+		CheckDestroy: testPkiSecretBackendRoleCheckDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testPkiSecretBackendRoleConfig_basic(name, backend, 3600, 7200, testLegacyPolicyIdentifiers),
+				Check: resource.ComposeTestCheckFunc(
+					append(checks,
+						resource.TestCheckResourceAttr(resourceName, "policy_identifiers.#", "1"),
+						resource.TestCheckResourceAttr(resourceName, "policy_identifiers.0", "1.2.3.4"),
+					)...,
+				),
+			},
+			{
+				Config: testPkiSecretBackendRoleConfig_basic(name, backend, 3600, 7200, newPolicyIdentifiers),
+				Check: resource.ComposeTestCheckFunc(
+					append(checks,
+						resource.TestCheckResourceAttr(resourceName, "policy_identifier.#", "2"),
+						resource.TestCheckTypeSetElemNestedAttrs(resourceName, "policy_identifier.*", map[string]string{"oid": "1.2.3.4.5", "cps": "https://example.com/cps", "notice": "Some notice"}),
+						resource.TestCheckTypeSetElemNestedAttrs(resourceName, "policy_identifier.*", map[string]string{"oid": "1.2.3.4.5.6"}),
+					)...,
+				),
+			},
+		},
```

Comment:
We might want test that the two fields are mutually exclusive by adding a `ExpectError`  `TestStep` similar to:

https://github.com/hashicorp/terraform-provider-vault/blob/70a187215f580d6dae538d9eb3c0e17540ac337a/vault/data_source_gcp_auth_backend_role_test.go#L99-L104

Reply:
Ah, perfect. I had wanted to do that, but didn't quite know how. That helps.